### PR TITLE
Add IPPREFIX to/from VARCHAR casting

### PR DESCRIPTION
**Pull request by mohsaka**
Add IPPREFIX to/from VARCHAR casting.

Fuzzer changes here:
https://github.com/facebookincubator/velox/pull/11209

### CODE REVIEW

File: velox/functions/prestosql/TypeOf.cpp [hunk 1]
```diff
@@ -93,6 +91,9 @@ std::string typeName(const TypePtr& type) {
           typeName(type->childAt(0)),
           typeName(type->childAt(1)));
     case TypeKind::ROW: {
+      if (isIPPrefixType(type)) {
```
Comment (aditi-pandit) [inline velox/functions/prestosql/TypeOf.cpp:94]: @mohsaka : The change in this file should also be submit as an individual PR.

File: velox/functions/prestosql/types/IPAddressType.cpp [hunk 2]
```diff
@@ -57,7 +57,7 @@ class IPAddressCastOperator : public exec::CastOperator {
       castFromVarbinary(input, context, rows, *result);
     } else {
       VELOX_UNSUPPORTED(
-          "Cast from {} to IPAddress not supported", resultType->toString());
+          "Cast from {} to IPAddress not supported", input.type()->toString());
```
[the remark this answers is not in the document]
Comment (mohsaka) [reply velox/functions/prestosql/types/IPAddressType.cpp:60]: https://github.com/facebookincubator/velox/pull/11448

File: velox/functions/prestosql/TypeOf.cpp (hunk 1 above)
Comment (mohsaka) [reply velox/functions/prestosql/TypeOf.cpp:94]: https://github.com/facebookincubator/velox/pull/11449

File: velox/functions/prestosql/types/IPPrefixType.cpp [hunk 3]
```diff
@@ -51,8 +80,114 @@ class IPPrefixCastOperator : public exec::CastOperator {
...
+
+      addrBytes = (ip.isIPv4Mapped() || ip.isV4())
+          ? folly::IPAddress::createIPv4(ip)
+                .mask(prefix)
+                .createIPv6()
+                .toByteArray()
+          : folly::IPAddress::createIPv6(ip).mask(prefix).toByteArray();
+
+      std::reverse(addrBytes.begin(), addrBytes.end());
+      memcpy(&intAddr, &addrBytes, ipaddress::kIPAddressBytes);
```
Comment (czentgr) [inline velox/functions/prestosql/types/IPPrefixType.cpp:187]: Can we add a compile time check here that `sizeof(intAddr)` and `ipaddress::kIPAddressBytes` are the same?

File: velox/functions/prestosql/types/IPPrefixType.cpp [hunk 4]
```diff
@@ -51,8 +80,114 @@ class IPPrefixCastOperator : public exec::CastOperator {
...
+    auto* flatResult = result.as<FlatVector<StringView>>();
+    const auto* ipprefixes = input.as<RowVector>();
+    const auto* ip = ipprefixes->childAt(0)->as<SimpleVector<int128_t>>();
+    const auto* prefix = ipprefixes->childAt(1)->as<SimpleVector<int8_t>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto intAddr = ip->valueAt(row);
+      folly::ByteArray16 addrBytes;
+
+      memcpy(&addrBytes, &intAddr, ipaddress::kIPAddressBytes);
```
Comment (czentgr) [inline velox/functions/prestosql/types/IPPrefixType.cpp:107]: Can we add a compile time check here that the `sizeof(addrBytes)` and `ipaddress::kIPAddressBytes` are the same?

**Pull request closed without merging**